docs/design/sync-manager.md

```diff
@@ -269,13 +273,181 @@ manager, we have the following hard requirements:
+
+After a sync, the sync manager will return the list of known engines with the
+same tri-state value. In almost all cases, the list of engines returned will
+have either "Allowed" or "Declined" - "Unknown" is an edge case (for example, if
+you invent an engine name, you'll get this back.) Importantly, the application
+must expect that the engines it gets back after the sync might include engines
+that were not initially specified - it will get the *complete* set of known
+engines and their state. The application is expected to persist these values and
+supply them to subsequent `sync()` calls, although it need not provide UI
+allowing them to be changed. Note that it technically could supply such UI,
+but it is advised not to, for UX reasons discussed further below.
+
+On the very first sync for an application (eg, immediately after being
+connected to an account), and assuming the application has not explicitly
+asked what to sync, the application should supply the list of engines
+with "Unknown" for the state values. However, if the application has asked
+the user to choose, it is free to specify "Allowed" or "Declined" - it just
+needs to be aware that doing so may overwrite current values for the account.
+
+If a device supplies "Unknown" for an engine, and there's no storage for the
+account (eg, a device syncing after a node reassignment) it will not sync that
+engine. This is because it doesn't know what the state for that engine should
+be. It will start syncing the engine when some other device shows up and either
+adds the collection or adds it to declined, or if the user explicitly visits
+"choose what to sync" on the device. This might be surprising, and possibly
+contentious, but should be very rare.
+
+#### Examples
+
+(Note that for simplicity, this section uses JSON to represent the declined
+list, but that's not how it will be specified in practice)
+
+Let's take Fenix as an example, when a user signs in to an existing account
+and before they've explicitly chosen what to sync on that device:
+
+* On the very first sync, it will supply `{"bookmarks": Unknown, "history", Unknown}`
+* When sync returns, it might get back, for example.
+  `{"bookmarks": Enabled, "history": Disabled, "passwords": Enabled, ...}`
+* The application is expected to persist *all* of these values - including 
+  "passwords", even though it doesn't understand that collection. This value
+  may be used to repopulate the declined list should this device be the first to
+  sync after storage is reset.
+
+For any app on which an account is created, and where a "choose what to sync"
+UI is part of the account creation flow:
+
+* On the very first sync, it will supply all of the "offered" engines, and
+  will supply explicit "Enabled" or "Disabled" values - nothing will be specified
+  as "Unknown".
+* Because this will be the first sync for the account, no other engines will
+  exist on the server, so the exact same states will be returned after the sync.
```

Comment:
not clear this is true: consider acct creation on a device which doesn't support "passwords" although it is offered. After the sync passwords will not be in declined, but also will not exist as a collection, so will be returned as Unknown. From the above rules, passwords will be declined as there's nowhere to record the acceptance :(

Do we need to bite-the-bullet and add `accepted` as a sibling of `declined` and thus ignore the state of info/collections in all the above?

Reply:
I guess I was thinking of when, say, CWTS is offered by the server as part of account creation. Ryan [has a doc](https://docs.google.com/document/d/1jiaVYI1i1JS31ZEKjbzLzexK5TJBrYKO58dcl5wynwI) talking about how account creation isn't the right place and it's going to get "more wrong" as accounts are created for non-sync purposes - however, he still proposes offering CWTS exactly once - so I guess I can see a future where we decide to offer "all" collections even if the current app only supports a subset.

Not clear though - it's difficult to predict future UX asks, should we probably should *allow* for this capability?

Reply:
Did we settle on adding `accepted` in the end?